src/chart/generateCategoricalChart.tsx

```diff
@@ -707,6 +719,7 @@ const calculateOffset = (
   }
 
   if (legendItem && prevLegendBBox) {
+    // @ts-expect-error margin is optional in props but required in appendOffsetOfLegend
     offset = appendOffsetOfLegend(offset, graphicalItems, props, prevLegendBBox);
```

Comment:
what happens when margin isn't provided 😅? wouldn't this have caused runtime errors in the `appendOffsetOfLegend` calculations below?

Reply:
I think the whole chart just won't render without errors. Not sure where that's coming from. Definitely suspicious.

Reply:
Hmm - I think there is default margin? Maybe it's optional but defaulted